packages/components/src/components/editor/graph/graph.tsx

```diff
@@ -0,0 +1,119 @@
+import '@xyflow/react/dist/style.css';
+import { Button } from '@/components/common/button/button';
+import { Flex } from '@/components/common/flex/flex';
+import { CustomNode } from '@/components/editor/graph/nodes/customNode';
+import FloatingEdge from '@/components/editor/graph/edges/floatingEdge';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { Background, ConnectionMode, Controls, MiniMap, Panel, ReactFlow, ReactFlowProvider, type Node } from '@xyflow/react';
+import { BasicSelect } from '@/components/common/select/select';
+import CircleFloatingEdge from '@/components/editor/graph/edges/circleFloatingEdge';
+import FloatingConnectionLine from '@/components/editor/graph/edges/FloatingConnectionLine';
+import useCustomGraph from '@/components/editor/graph/data/useCustomGraph';
+
+export type CustomNodeData = {
+  id: string;
+  label: string;
+  info?: string;
+  content?: React.ReactNode;
+  target?: Array<EdgeData>;
+  options?: {
+    highlightNode?: boolean;
+    disableHandles?: boolean;
+    controls?: React.ReactNode;
+    expandContent?: boolean;
+  };
+};
+
+export type EdgeData = {
+  id: string;
+  label?: string;
+};
+
+export type GraphNode = Node<{ CustomNodeData: CustomNodeData }, 'custom'>;
```

Comment:
why is this object key uppercase?
```suggestion
export type GraphNode = Node<{ customNodeData: CustomNodeData }, 'custom'>;
```